lib/ansible/modules/network/nxos/nxos_interface.py

```diff
@@ -484,10 +473,12 @@ def map_obj_to_commands(updates, module):
 
             else:
                 commands.append(interface)
-                if mode == 'layer2':
-                    commands.append('switchport')
-                elif mode == 'layer3':
-                    commands.append('no switchport')
+                # Don't run switchport command for loopback and svi interfaces
+                if interface_type in ('ethernet', 'portchannel'):
```

Comment:
Please change this to `get_interface_type(name)` here. `interface_type` is one of the module parameters which is mutually exclusive to module param `name` or `interface`. `get_interface_type` function returns type of the interface.